app/javascript/src/Common/components/SelectWithModal.tsx

```diff
@@ -89,7 +86,7 @@ const SelectWithModal = <T extends IRecord>({
     if (pageIsEmpty && thereAreMoreItems) {
       setIsLoading(true)
 
-      fetchItems({ page, perPage: PER_PAGE })
+      fetchItems({ page, perPage: PER_PAGE, query })
```

Comment:
So, this `useEffect` is called when going through pages.

So, imagine the situation: the user types a keyword, and clicks "Search". The first 20 results are fetched (through the `useEffect` with `query` dependency), which is 4 pages of 5 results. Let's imagine there are more results, so the user clicks on "next page". This is where this line is executed. And without `query` in the params, the 5th page will not show the results filtered by the keyword, so it was broken.